core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -434,12 +434,14 @@ public String toString() {
      */
     @VisibleForTesting
     public void solve() {
+        if (Context.get().isRelaxProofOfWork())
+            return;
```

Comment:
It's also nice to avoid short-circuiting returns.

Reply:
I love them (-: